test/cmd/lfstest-gitserver.go

```diff
@@ -481,6 +497,98 @@ func storageHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		w.WriteHeader(404)
+	case "HEAD":
+		// tus.io
+		if len(r.Header.Get("Tus-Resumable")) == 0 {
+			log.Fatal("Missing Tus-Resumable header in request")
+			w.WriteHeader(400)
+			return
+		}
+		parts := strings.Split(r.URL.Path, "/")
+		oid := parts[len(parts)-1]
+		if by, ok := largeObjects.GetIncomplete(repo, oid); ok {
+			w.Header().Set("Upload-Offset", strconv.FormatInt(int64(len(by)), 10))
+		} else {
+			w.Header().Set("Upload-Offset", "0")
+		}
+		w.WriteHeader(200)
+	case "PATCH":
+		// tus.io
+		if len(r.Header.Get("Tus-Resumable")) == 0 {
```

Comment:
Not a big deal, but perhaps you could extract this to:

```go
func AssertTusResumableHeader(r *http.Request, w http.ResponseWriter) (present bool) {
        if len(r.Header.Get("Tus-Resumable")) == 0 {
                log.Fatal("Missing Tus-Resumable header in request")
                w.WriteHeader(http.StatusBadRequest)

                return false
        }

        return true
}
```

and use that? That method is sort of side-effect-y, but I'm generally not as concerned about that in the testserver code.